plugins/bbs_exporter.js

```diff
@@ -0,0 +1,368 @@
+(function() {
+    var electron = require("electron");
+    var button;
+    var lastOptions = {};
+    var sides = {
+        north: "front",
+        south: "back",
+        west: "left",
+        east: "right",
+        up: "top",
+        down: "bottom"
+    };
+
+    /* Model exporter code */
+
+    function createHierarchy(objects)
+    {
+        var groups = {};
+
+        for (let i = 0; i < objects.length; i++)
+        {
+            var object = objects[i];
+
+            if (object.type === "group")
+            {
+                groups[object.name] = createModelGroup(object, groups);
+            }
+        }
+
+        return groups;
+    }
+
+    function createModelGroup(o, groups)
+    {
+        var group = {};
+        var cubes = [];
+
+        group.origin = o.origin.slice();
+
+        if (!o.rotation.allEqual(0))
+        {
+            group.rotate = o.rotation.slice();
+        }
+
+        if (typeof o.parent.name === "string")
+        {
+            group.parent = o.parent.name;
+        }
+
+        for (let i = 0; i < o.children.length; i++)
+        {
+            var object = o.children[i];
+
+            if (object.type === "group")
+            {
+                groups[object.name] = createModelGroup(object, groups);
+            }
+            else
```

Comment:
You are forgetting to check if the element is a cube. It currently throws errors when the model contains locators etc.

Reply:
I added `.type == "cube"` check, thank you! Is it good now?